tests/TestCase/ORM/MarshallerTest.php

```diff
@@ -558,6 +558,30 @@ public function testOneBelongsToManyJoinDataAssociatedWithIds()
     }
 
     /**
+     * Test HasMany association with _ids attribute
+     *
+     * @return void
+     */
+    public function testHasManyWithIds()
+    {
+        $data = [
+            'username' => 'lux',
+            'password' => 'passphrase',
+            'comments' => [
+                '_ids' => [1, 2]
+            ]
+        ];
+
+        $userTable = TableRegistry::get('Users');
+        $userTable->hasMany('Comments');
+        $commentTable = TableRegistry::get('Comments');
+        $user = $userTable->newEntity($data, ['associated' => ['Comments']]);
+
+        $this->assertEquals($user->comments[0], $commentTable->get(1));
+        $this->assertEquals($user->comments[1], $commentTable->get(2));
```

Comment:
After our short conversation I could understand what you were asking me to do. It is done!
Thanks